index.js

```diff
@@ -29,13 +29,12 @@ function fast (functions, context) {
     var that = this
 
     this._next = function () {
-      var fn = that.functions[that.i++]
-      if (!fn) {
+      if (that.i === that.functions.length) {
```

Comment:
Nice catch! :)
Can we cache the functions length outside this function? We should gain even a little more :)